standardtracer/tracer.go

```diff
@@ -2,16 +2,67 @@ package standardtracer
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
 
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-// New creates and returns a standard Tracer which defers to `recorder` after
-// RawSpans have been assembled.
-func New(recorder SpanRecorder) opentracing.Tracer {
+type sampleFunc func(int64) bool
```

Comment:
in our implementation Sampler is an interface, similar to the Reporter (or Recorder), because we want them to have a `Close()` method. One of our Sampler implementations is called "remotely controlled", because it establishes a connection to the mothership, which needs to be closed for clean shutdown.